__init__.py

```diff
@@ -1,121 +1,30 @@
-from anki.media import *
-from anki.notes import Note
-from aqt import mw, dialogs
-from anki.find import Finder
+from concurrent.futures import Future
+from aqt.mediacheck import MediaChecker
+from aqt import mw
+from aqt.utils import tooltip
 
-# oldCheck = MediaManager.check
-# def check(self, local=None):
-#     (missingFiles, unusedFiles) = oldCheck(self, local)
-#     nids = set()
-#     query = """select id from notes where flds like "" """
+TAGNAME = "MissingMedia"
 
-def check(self, local=None):
-    "Return (missingFiles, unusedFiles)."
-    mdir = self.dir()
-    # gather all media references in NFC form
-    allRefs = set()
-    refsToNid = dict() # this dic is new
-    for nid, mid, flds in self.col.db.execute("select id, mid, flds from notes"):
-        noteRefs = self.filesInStr(mid, flds)
-        # check the refs are in NFC
-        for f in noteRefs:
-            # if they're not, we'll need to fix them first
-            if f != unicodedata.normalize("NFC", f):
-                self._normalizeNoteRefs(nid)
-                noteRefs = self.filesInStr(mid, flds)
-                break
-        # new. update refsToNid
-        for f in noteRefs:
-            if f not in refsToNid:
-                refsToNid[f] = set()
-            refsToNid[f].add(nid)
-        # end new
-        allRefs.update(noteRefs)
-    # loop through media folder
-    unused = []
-    if local is None:
-        files = os.listdir(mdir)
-    else:
-        files = local
-    renamedFiles = False
-    dirFound = False
-    warnings = []
-    for file in files:
-        if not local:
-            if not os.path.isfile(file):
-                # ignore directories
-                dirFound = True
-                continue
-        if file.startswith("_"):
-            # leading _ says to ignore file
-            continue
 
-        if self.hasIllegal(file):
-            name = file.encode(sys.getfilesystemencoding(), errors="replace")
-            name = str(name, sys.getfilesystemencoding())
-            warnings.append(
-                _("Invalid file name, please rename: %s") % name)
-            continue
+def on_finished(self: MediaChecker, fut: Future) -> None:
+    self._on_finished(fut)
```

Comment:
since we don't use future here, I'd go with `*args, **kwargs`, so that it does not have to change with future API change. But really implementor choice here

Reply:
We're using future to get Check Media's output here.